### PR TITLE
[gas-calibration][simple] ignore terms with 0-coefficients

### PR DESCRIPTION
**Pull request by vgao1996**
This revises `fn aggregate_terms` so that it ignores all terms with 0 as coefficients, preventing them from causing errors in the gas calibration tool downstream.

### CODE REVIEW

**Review by wlawt — APPROVED** [review]

File: aptos-move/aptos-abstract-gas-usage/src/algebra_helpers.rs [hunk 1]
```diff
@@ -79,46 +79,48 @@ pub fn expand_terms(expr: DynamicExpression) -> Vec<DynamicExpression> {
 ///
 /// (A + A) * (5 + 5) => 5A + 5A + 5A + 5A => 20A
 pub fn aggregate_terms(terms: Vec<DynamicExpression>) -> Result<BTreeMap<String, u64>> {
+    use DynamicExpression::*;
```
Comment (brmataptos) [inline aptos-move/aptos-abstract-gas-usage/src/algebra_helpers.rs:82]: Usually it's better to enumerate what you use, so changes don't break your code.

File: aptos-move/aptos-abstract-gas-usage/src/algebra_helpers.rs [hunk 2]
```diff
@@ -79,46 +79,48 @@ pub fn expand_terms(expr: DynamicExpression) -> Vec<DynamicExpression> {
...
+                // Seeing a concrete quantity means that the user is not encoding the gas
+                // cost in an abstract way using GasExpression.
+                //
+                // This makes calibration impossible and we should error.
                 if value != 0 {
-                    return Err(anyhow!(
-                        "Concrete quantity provided. Should be GasExpression."
-                    ));
+                    bail!(
+                        "Expected simple term got non-zero concrete value {}.",
```
Comment (brmataptos) [inline aptos-move/aptos-abstract-gas-usage/src/algebra_helpers.rs:94]: Do we all know what "simple term" means?  I assume it's (constant * var) ?

**Review by brmataptos — APPROVED** [review]

File: aptos-move/aptos-abstract-gas-usage/src/algebra_helpers.rs (hunk 2 above)
Comment (wlawt) [reply aptos-move/aptos-abstract-gas-usage/src/algebra_helpers.rs:94]: Agreed, we should probably define "simple term" for clarity

File: aptos-move/aptos-abstract-gas-usage/src/algebra_helpers.rs (hunk 1 above)
Comment (davidiw) [reply aptos-move/aptos-abstract-gas-usage/src/algebra_helpers.rs:82]: agreed, I don't see how this is better.

**Review by davidiw — APPROVED** [review]

File: aptos-move/aptos-abstract-gas-usage/src/algebra_helpers.rs (hunk 1 above)
Comment (vgao1996) [reply aptos-move/aptos-abstract-gas-usage/src/algebra_helpers.rs]: I now manually import all the variants. Thanks for pointing this out!

File: aptos-move/aptos-abstract-gas-usage/src/algebra_helpers.rs (hunk 2 above)
Comment (vgao1996) [reply aptos-move/aptos-abstract-gas-usage/src/algebra_helpers.rs:90]: Revised the comments and now they clearly explain what a simple term means.

**Comment by vgao1996** [comment]
@davidiw @brmataptos I've made the changes you requested and cleaned up the code a bit, getting rid of some unnecessary allocations.

**Pull request merged**